File: MatrixMultiplikation_java/src/matrixmult/MatrixThreadrunner.java
package matrixmult;

public class MatrixThreadRunner implements Runnable {
    private Matrix matrixA;
    private Matrix matrixB;
    private Matrix matrixC;
    private int upperBound;
    private int lowerBound;

    MatrixThreadRunner(Matrix matrixA, Matrix matrixB, Matrix matrixC, int upperBound, int lowerBound) {
        this.matrixA = matrixA;
        this.matrixB = matrixB;
        this.upperBound = upperBound;
        this.lowerBound = lowerBound;
        this.matrixC = matrixC;
    }

    public void run() {
        try {
            for (int i = lowerBound; i < upperBound; i++) {
                for (int k = 0; k < matrixB.getMatrix().length; k++) {
                    for (int j = 0; j < matrixA.getMatrix().length; j++) {
                        matrixC.getMatrix()[i][k] += matrixA.getMatrix()[i][j] * matrixB.getMatrix()[j][k];
                    }
                }
            }
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}

File: MatrixMultiplikation_Kotlin/src/MatrixThreadRunner.kt
class MatrixThreadRunner(private val matrixA: Matrix, private val matrixB: Matrix, private val upperBound: Int, private val lowerBound: Int) : Runnable {
    override fun run() {
        try {
            val matrixC = Matrix(matrixA.matrix.size, matrixA.matrix[0].size)
            for (i in lowerBound until upperBound) {
                for (k in matrixB.matrix.indices) {
                    for (j in matrixA.matrix.indices)
                        matrixC.matrix[i][k] += matrixA.matrix[i][j] * matrixB.matrix[j][k]
                }
            }
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }
}